### PR TITLE
Port gui

### PR DESCRIPTION
**Pull request by jamesblack3**
- Uncommented the file
- Changed GlStateManager to RenderSystem where possible
- Changed the scale factor calculation to use the MainWindow's getGuiScaleFactor function
- The tex function in the BufferBuilder uses floats now instead of doubles
- RenderSystem uses color4f instead of color
- Some of the RenderSystem functions names changed and are deprecated, may want to look at those at a later date
- Button.id doesn't exist anymore. Need to use IPressable in the constructor of buttons. Not sure if this file needs it or if that can be done instead by the other modules that are creating the buttons

### CODE REVIEW

File: src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java [hunk 1]
```diff
@@ -1,162 +1,149 @@
...
+
+	public void setShowSelectionBox(boolean val) {
+		showSelectionBox = val;
+	}
+
+	protected int getContentHeight() {
+		return getNumElements() * slotHeight;
+	}
+//  // b.id doesn't exist, no need for this either then
 //  public void setScrollButtonIds(int scrollUpButtonID, int scrollDownButtonID) {
```
Comment (Rover656) [inline src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java:122]: What does this do; I'm sure its needed for something?

File: src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java [hunk 2]
```diff
@@ -1,162 +1,149 @@
...
+			amountScrolled = i;
+		}
+	}
+
+	public int getContentOverhang() {
+		return getContentHeight() - (height - margin);
+	}
+
+//  // b.id doesn't exist anymore, need to use IPressable
 //  public void actionPerformed(@Nonnull Button b) {
```
Comment (Rover656) [inline src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java:145]: again, should probably be done as part of this PR.

**Review by Rover656 — CHANGES_REQUESTED** [review]
I trust that all the rendering stuff will work fine; I'm not an expert here so I'll allow anyone else to chip in. I just think these couple areas should be addressed to make these complete.

File: src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java (hunk 1 above)
Comment (jamesblack3) [reply src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java:122]: I'm not sure. It was already there so I just ported it over.

File: src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java (hunk 2 above)
Comment (jamesblack3) [reply src/main/java/com/enderio/core/client/gui/widget/ScrollableListWidget.java:145]: Are you referring to line 145? The preview isn't clear...

Well if we need to use IPressable in the constructor of the buttons, then I was having trouble finding where the buttons are being made. The original code seems to have the buttons made elsewhere and their IDs get passed in via setScrollButtonIds.

**Comment by ferriarnus** [comment]
@jamesblack3 are you still working on it? Otherwise I'll give it a go.

**Comment by jamesblack3** [comment]
Go ahead, I wasn't really sure what I was doing.

**Comment by Macleykun** [comment]
> Go ahead, I wasn't really sure what I was doing.

perhaps you can better close the pull?

**Pull request closed without merging**